### PR TITLE
[RDY] Fix CMake 3.5 support and verify with CI

### LINKED ISSUE #1780: [Bug] Can't compile, RncLib warning/error
#### Describe the issue

Following the instructions, I attempted to compile the current git source, but I get an error related to RncLib, and it fails:

squigley@Latitude-E5530-non-vPro ~/CorsixTH $ mkdir build
squigley@Latitude-E5530-non-vPro ~/CorsixTH $ cd build
squigley@Latitude-E5530-non-vPro ~/CorsixTH/build $ cmake -DUSE_SOURCE_DATADIRS=ON .. 
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Note: SDL audio is enabled (default)
Note: FFmpeg video is enabled (default)
Note: FreeType2 is enabled (default)

Building common libraries
You have called ADD_LIBRARY for library RncLib without any source files. This typically indicates a problem with your CMakeLists.txt file
Building CorsixTH
CMake Error at CorsixTH/CMakeLists.txt:46 (add_executable):
  add_executable called with incorrect number of arguments

CMake Error at CorsixTH/CMakeLists.txt:57 (target_link_libraries):
  Cannot specify link libraries for target "CorsixTH" which is not built by
  this project.

-- Configuring incomplete, errors occurred!
See also "/home/squigley/CorsixTH/build/CMakeFiles/CMakeOutput.log".

I found the files under libs/rnc, and they don't configure correctly directly either:

squigley@Latitude-E5530-non-vPro ~/CorsixTH/libs/rnc $ cmake .
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
You have called ADD_LIBRARY for library RncLib without any source files. This typically indicates a problem with your CMakeLists.txt file
CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.10)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
-- Build files have been written to: /home/squigley/CorsixTH/libs/rnc

#### Steps to Reproduce
1. git clone the source

2. mkdir build

3. cd build
4. cmake -DUSE_SOURCE_DATADIRS=ON ..

#### Expected Behaviour

cmake would complete, so I could compile the source.

#### System Information
**CorsixTH Version: git as of Dec 26th** file in .git includes " 541540bc9f95719269d9129196f53493ef3e988f Simon Quigley <squigley@Latitude-E5530-non-vPro.(none)> 1609043185 -0500	clone: from https://github.com/CorsixTH/CorsixTH.git"

**Operating System: Linux - 4.15.0-128-generic - x86_64** (Mint 19)

gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04) 

### Additional Info
*Paste any screen shots or other additional information that might help
illustrate the problem.*
[CMakeOutput.log](https://github.com/CorsixTH/CorsixTH/files/5748668/CMakeOutput.log)

### PR DESCRIPTION
**Pull request by TheCycoONE**
*Fixes #1780*

Use older cmake syntax for add_library in RncLib for compatibility with advertised version.

### CODE REVIEW

**Comment by TheCycoONE** [comment]
If someone actually has 3.5 or made a CI that ran with 3.5 I would feel more comfortable advertising we support it.

**Comment by lewri** [comment]
@tobylane might be able to double check?

**Comment by TheCycoONE** [comment]
CI now has cmake 3.5 enabled and has indicated there are other issues to be fixed.

File: .github/workflows/Linux.yml [hunk 1]
```diff
@@ -14,19 +14,33 @@ jobs:
...
+            docs: 0
+            cmake:
+            lua: Lua 5.1
             luac: luac5.1
             packages: liblua5.1-dev lua5.1
-          - lua: Lua 5.3
+          - static_analysis: 1
+            docs: 1
+            cmake: 3.5
+            type: full
```
Comment (tobylane) [inline .github/workflows/Linux.yml:33]: matrix.type isn't used.

**Review by tobylane — COMMENTED** [review]
Other than type, all good.

File: .github/workflows/Linux.yml (hunk 1 above)
Comment (TheCycoONE) [reply .github/workflows/Linux.yml]: gone

**Comment by squigley** [comment]
I think this was in response to my bug ticket of not being able to configure/compile it (on Linux Mint 19), and after this was changed, it now configures and compiles properly, in case that helps @TheCycoONE

**Pull request merged**